cranelift/wasm/src/sections_translator.rs

```diff
@@ -27,9 +27,7 @@ use wasmtime_types::ConstExpr;
 
 fn tag(e: TagType) -> Tag {
     match e.kind {
-        wasmparser::TagKind::Exception => Tag {
-            ty: TypeIndex::from_u32(e.func_type_idx),
-        },
+        wasmparser::TagKind::Exception => Tag::partial(e.func_type_idx),
```

Comment:
This seems to be the only user of `Tag::partial`, and the `tag` function here is only called from `parse_import_section` in the same file. The result is then passed to `ModuleEnvironment::declare_tag_import`. The latter unconditionally fails with an error.

So it seems that some related code here is unimplemented, which means that we could just fail in the `tag` function here already for the time being. Then we don't need `Tag::partial` at all.

Reply:
`Tag::partial` is used twice. See `section_translator.rs` in cranelift.

Reply:
I regard `Tag::partial` as a temporary hack. It will change eventually when upstream catches up.

Reply:
Okay, I overlooked the second use of `partial`, so it seems not as easy to get rid of as I hoped